### PR TITLE
MB-53 Documentation Updates for USWDS 2.0

### PR DESCRIPTION
**Pull request by gidjin**
## Description

This pull request adds some documentation around using USWDS 2.0, it also links to a detailed article on the differences between USWDS 1.0 and 2.0. It turns off the `USWDS Theme Notifications` to avoid extra noise in our build logs that might confuse people debugging issues. The documentation around our JavaScript testing frameworks has been updated and reworded to be clearer. Updates to the `docs/how-to/run-storybook.md` docs to make it clearer and link to detailed installation instructions.

This PR also includes the installation of [Danger JS](https://danger.systems/js) to run additional checks. These checks will be run in the new CircleCI job `dangerjs_checks`

New Danger Checks
* Warning if your PR description is too short (less than 10 characters)
* Warning if your PR has javascript app code changes but no test changes
* Failure if your PR introduces new files in `src/components` but no modifications of Storybook stories in `src/stories`
* Warning if your PR changes `package.json` but not `yarn.lock`

This PR changes the `no-debugger` to a warning vs error to allow `yarn build` locally to pass so that debugger can be used in testing locally.

This PR updates `make storybook_tests` to check for port 6006 to be open, if not offer to shut down the process taking up port 6006 or abort the script.

## Reviewer Notes

Please review the documentation for clarity, spelling errors, or improvements. If you could run the commands in setup to ensure they run for you locally, that would also be helpful.

## Setup

If you have [Grip](https://github.com/joeyespo/grip) you can run `grip` from the root of the project then view the frontend documentation

* http://localhost:6419/docs/frontend.md
* http://localhost:6419/docs/how-to/run-storybook.md

New or updated commands

Be sure you run `yarn install` before trying the following

* `make storybook_tests`
  * If you run the above after running `yarn storybook` in another terminal you will see the prompt about port 6006 in use
* `yarn danger local` - will run danger checks against your local repo. 

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [MB-53](https://dp3.atlassian.net/browse/MB-53) for this change

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -902,6 +902,19 @@ jobs:
...
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-cache-yarn-v2-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          keys:
+            - v2-mymove-node-modules-{{ checksum "yarn.lock" }}
+      - run: yarn danger ci
+      - announce_failure
```
Comment (chrisgilmerproj) [inline .circleci/config.yml]: Could we do this in the `client_test` step? Every time we start up a new executor it costs a minimum 200 CircleCI credits. So if its not a long check then let's put it in something that already exists for JS.

File: src/shared/styles/_settings.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
+/* reduce noise in logs by turning off uswds notifications */
+$theme-show-notifications: false;
```
Comment (chrisgilmerproj) [inline src/shared/styles/_settings.scss]: Does this disable the giant notification that prints when building? if so, thank you!

Comment (gidjin) [reply src/shared/styles/_settings.scss]: Yes it does, you're welcome 🙂

File: .circleci/config.yml (hunk 1 above)
Comment (gidjin) [reply .circleci/config.yml]: Sure, I didn't realize it cost so much. My one hesitation is that danger can be used to check other things not just for the client but there might be a better job to join it too.

Comment (chrisgilmerproj) [reply .circleci/config.yml]: Another option is to make Danger a pre-commit hook and then you can run it in the `pre_test` job.  I have plans to have us combine other jobs like the acceptance test jobs and the build jobs. Really the fewer jobs we run in CircleCI the cheaper it becomes to run.

Comment (gidjin) [reply .circleci/config.yml]: I've made it part of `pre-test` as that made the most sense to me. It doesn't take long to run so it shouldn't make it much longer. I'll have better numbers once CI runs on my branch.

File: .circleci/config.yml [hunk 3]
```diff
@@ -435,7 +435,11 @@ jobs:
             mkdir -p tmp/test-results/pretest
             pre-commit run -v --all-files golangci-lint | tee tmp/test-results/pretest/golangci-lint.out
       # adding danger checks here as they are similar to our pre-commit hooks
-      - run: yarn danger ci --failOnErrors
+      - run:
+          name: Run DangerJS checks
+          command: |
+            echo 'export DANGER_GITHUB_API_TOKEN=1' >> $BASH_ENV
```
Comment (chrisgilmerproj) [inline .circleci/config.yml]: Usually after this we'd also have `source $BASH_ENV` to ensure that it is there.

Comment (gidjin) [reply .circleci/config.yml]: You are too fast my friend, I've switched it to use the environment tag. Also, I just commented it out I can't seem to find a way to run `yarn danger ci` without the token.

**Pull request merged**